### ISSUE TITLE
Issue with sub accounts using Parent account activities- possible missing revenue

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#1604]
In GitLab by @kdjstudios on Dec 24, 2019, 10:23

**Submitted by:** "Leah Mitchell" <leah.mitchell@answernet.com>

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/2019-12-23-92339

**Server:** Internal

**Client/Site:** Billings

**Account:** 071-1239

**Issue:** 

We have a parent/child group of accounts in Billings that don’t appear to be accumulating the minute usage from the child accounts into the Parent account invoices as of 10/1 (and the 11/1, and 12/1 invoices)

071-1239 Cummins Sales and Service Cedar Rapids is the parent account, but when we look at each sub account under it, the check boxes for ‘use parent activities’ and ‘Separate child account transactions’ are not checked.  When we try to check the box, it says the action is not applicable to a child account. 

However, when we remove the associated parent account in the ‘parent account’ box higher up on the screen, then those check boxes for ‘use parent account activities’ and ‘separate child account transactions’ automatically check themselves back off without being touched.

 

Juan also recalls that there used to be a ‘bill to’ checkbox in that area that appears to be gone now.  When you hover over the question mark for more details on the ‘use parent account activities’ checkbox, it says “this is not usable when both Bill To and Separate are not checked”.

Could the issue be related to the fact that the ‘bill to’ checkbox is no longer there?

 

I also have an example of this exact same issue on an El Paso client: HCB

 

Can you help us understand if this is the setting that is preventing the usage from these sub accounts from accumulating on the parent account invoices?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 24, 2019, 10:25

I wrote:

> Thanks for reaching out, for future reference please be sure to use the help wiki in SA Billing along with the guides located both on the VCC Portal and in SA Billing. 
> 
> The parent account is where you specify the options to enable the 'use parent activities' and 'separate child transactions' and this will effect all child accounts under that parent account. Note the "bill to" option was relabeled to "use parent activities", I will look through the documentation to see where we still have the "bill to" wording and update it. Below is the section about parent/child accounts from the help wiki for further clarifications. Please review this and let us know if there are any questions or concerns. 

Haydee wrote:

> Thank you Kyle, we are going to go through the accounts and confirm what you are telling us is how we want them set up.  The problem is that it seems like it is now working like it should.


Leah wrote:

> Your explanations make sense, however, we do not believe it's working properly.  Here is the example of why we think that (was included in my original ticket as well).  Can you please check into this account for us or can we discuss together.
> 
> The parent account: 071-1239 Cummins Sales and Service Cedar Rapids *at the time of the December invoice and all prior to now (we increased their rate yesterday) had a base rate of $630 for 700 minutes with $.97/min overages.  The parent account is using code 55099 for usage.  
> 
>  I went through all 15 child accounts and added up the usage in 55099 on the human readable for each sub and the parent.  Here are the numbers I got which total 771- which means, they should have been billed for 71 excess minutes.  I have also attached the 12/1 invoice which shows no charges for excess at all.

![image](/uploads/5ca864049e1410febed0f8b55f4e5cde/image.png)

I wrote:

> We will have the support team confirm which options were enabled on the parent account at the time of billing in order to determine how the usage was applied to the activities and billed for. This will require them too look through the logs. We will follow up with their response accordingly.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 24, 2019, 10:32

Aavaz, please look into the account to see which options were enabled on the parent account at the time of billing and provide those. Please also provide which activities were active on both parent and child accounts at the time of billing.  From that information we should be able to determine why the client's invoice was generated the way it was.


NOTE: I did take a quick review of the accounts and here are my thoughts, but that is assuming the same options are enabled now that were before.
1. The parent account has "separate" and "use parent" enabled.
1. It appears there are no activities or recurring fees on the child accounts. 
1. I believe because they have separate enabled but do not have any activities on the child accounts, the usage from the child accounts is not being billed. 
1. I would think they will need to disable the "separate" option for the easy solution so all usage is billed to he parent account. Yet if they are needed to have each child account with their own allowances, then they need to keep "separate" enabled, but add the activities to each child account.

**Comment by kdjstudios** [comment]
In GitLab by @jredfield on Dec 29, 2019, 13:51

@ranand @smasih @kdjstudios Please make this high priority as we have not seen any updates from you regarding the analysis of the logs that we requested last Monday.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Dec 29, 2019, 22:15

assigned to @amishra

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Dec 30, 2019, 01:24

@kdjstudios @jredfield 

Thank you for the update.

We looked into this and our findings are described below:

* **For October(`10/1/19`), November(`11/1/19`), and December(`12/1/19`) Invoices**
  * Active Activity code was: `55099`
  * Excess rate was: `$0.97`
  * Allowance was: `700`
  * Allowance was greater than each quantity for all child/parent accounts. That's why they weren't applied to the invoices.

* **For January Invoice(`1/1/20`)** 
 * Active Activity code was: `55099`
  * Excess rate was: `$1.10`
  * Allowance was: `760`
  * Allowance was greater than each quantity for all child/parent accounts. That's why they weren't applied to the invoices.

**Note: As per my understanding, the functionality of parent/child is working correctly**

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 30, 2019, 09:12

I wrote:

> The support team was able to confirm that the allowance for the last few billing cycles was greater then the usage uploaded. Since the usage was less then the allowance there would be no excess charges on the invoice. Please see below.
> 
> For October(10/1/19), November(11/1/19), and December(12/1/19) Invoices
> - Active Activity code was: 55099
> - Excess rate was: $0.97
> - Allowance was: 700
> 
> For January Invoice(1/1/20)
> - Active Activity code was: 55099
> - Excess rate was: $1.10
> - Allowance was: 760
> 
> Please let us know if you have any questions or concerns.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 30, 2019, 09:16

@amishra 

Thanks I am glad the usage was less then the allowance, that is a relief that the parent/child is working correctly. However, after trying to look at the Usage/Fee history report on the account I was unable to confirm the total usage and it looks like these activities are completely missing from the report. I opened a new ticket to address this #1684

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 30, 2019, 10:04

Leah wrote:

> Unfortunately, that is not correct.  I mentioned in my first email that I added up all the minutes for the child accounts and they totaled 771, their allowance was 700 minutes, they should have been billed for 71 min overages.  

I wrote:

> We have brought this back to the support team to confirm the setup on the accounts and the actual usage that was applied to the activities. 


@amishra 

Please reconfirm the setup on the accounts and provide the usage that was then used for each billing cycle. Please keep in mind Leah is taking the total of all usage and then applying it to the parent account's activities. As stated above, I believe they need to have the "separate" not checked, in order to achieve this. http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1679#note_53461

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Dec 31, 2019, 06:09

@kdjstudios @jredfield 

Thank you for the update.

* I would like to inform you that we are not tracking every action when and who updates the account.

* but we looked into the invoice(**071-29388**) & invoice date `12/1/19` and found that there were **32** transactions line items which mean at the time of invoice creation they set **separate as true**. because if they set **separate as false** then they created only one transaction line item for the invoice.

**Please let us know if we missed something.**

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Dec 31, 2019, 06:09

added 3h of time spent at 2019-12-31

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 31, 2019, 10:22

I wrote:

> The support team looked further into this and determined that the parent account has "Separate Child Account Transactions" enabled, which is why the usage was being charged at the child account level and the child account activities. If you are wanting to sum the usage into the parent account please disable this setting on the parent account. Please also ensure you keep "Use parent account Activities" enabled on the parent account too.  This will result in all of the child account usage then being moved up to the parent account and charged based only upon the parent accounts activities. 
> 
> Please let us know if you have any questions or concerns.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 31, 2019, 13:33

Leah wrote:

> The client wants to see the child accounts separated on the invoice, that's why it's set up that way.  However, the minutes used by each child should roll up to the allotted min package on the parent.  How do we accomplish both?


@amishra 

From what I can tell is they need to have both checked in order to accomplish this, but it would appear that SA Billing is not functioning as it should. Here is the description from having both options checked.

> On the parent account, both “Use parent account activities” and “Separate child account transactions” are checked. In this case, the usage would be moved to the parent account like in example 1 and the line items would be separate on the invoice like in Example 2. 

Please look further into why the usage is not being moved to the parent account as it should. Because if it was, it should have generated a change on the invoice since "771" was the total usage, and as you stated the allowance was "700" for the 12/1 invoice.

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Jan 2, 2020, 04:35

@kdjstudios @jredfield 

> On the parent account, both “Use parent account activities” and “Separate child account transactions” are checked. In this case, the usage would be moved to the parent account like in example 1 and the line items would be separate on the invoice like in Example 2.

* Yes, you are right but at the time of invoice creation, both were true because the invoice has 32 line items.
* If separate was **false** then only one line item would have been created with the sum of all quantity as described above by Leah.

Although, We have tested on dev server as the same usage file(1/12).

CASE 1: Use Parent Account Activities is set as `True` and  Separate child accounts transactions is set as 'True'

In this case, it will sum all the parent & child account activities usage value and it will treat it as a single transaction. For Eg.

![image](/uploads/36ed7de0d28b9546931e5e9335a94663/image.png)

CASE 2: Use Parent Account Activities is set as `True` and  Separate child accounts transactions is set as 'False'

In this case, it will treat all the activities of the parent and child accounts as separate transactions.
But to calculate the quantity on the child accounts, it will take base rate and allowance of the parent account activities.

For Eg. 

* Account `071-1234` actual quantity is **193** but its allowance is **700**, so its billable is **0**
* Account `071-1237` actual quantity is **37** but its allowance is **700**, so its billable is **0**

![image](/uploads/af917637293952564d10830167cfb104/image.png)

**In short, they had separate `true` at the time of invoice creation.**

I hope this helps.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 3, 2020, 08:55

@ranand @amishra @smasih @jredfield 

What you described above is exactly what is not occurring and is contradicting what was stated in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1679#note_53735.  Please see ticket #1697 for another report of similar issues. 

Please see the below examples of how the charges should be applying. This was taken directly from the documentation created as part of #1362. 

Examples:

**• Example 1:** On the parent account, “Use parent account activities” is checked and “Separate child account transactions” is not. In this case the usage from the child accounts will be moved to the parent account, then combined and applied against the parent accounts charges.

**• Example 2:** On the parent account, “Use parent account activities” is not checked and “Separate child account transactions” is. In this case, the usage on the child accounts would have the charges applied, and then the invoice for the parent account would display each child account number, description and charges.

**• Example 3:** On the parent account, both “Use parent account activities” and “Separate child account transactions” are checked. In this case, the usage would be moved to the parent account like in example 1 and the line items would be separate on the invoice like in Example 2.

**• Example 4:** Neither “Use parent account activities” nor “Separate child account transactions” are checked. In this case, the usage would be charged at the child account level and then moved to the parent account. Because usage rates may vary from one child account to another, and to the parent when showing the invoice draft, it will show each line item separate but combined on the printed invoice. The draft invoice will look identical to the example 2. Allowing for editing of each line item to be edited at child account base level. The quantity will be the sum of the billable usage from all accounts that have that billable item. The amount will be the sum of the billable charge amount from all accounts that have that billable item (child and parent). Label on invoice: The label will be the description at the parent account if available. If not available, it will use the label at the site level. NOTE: We can not display the “usage breakdown on these accounts.


You stated the following, but that does not match with the documentation above "Example 1":

![image](/uploads/5ec4a80e2b363cf00d25b5a290a49aa1/image.png)

You stated the following, but that does not match with the documentation above in "Example 3":

![image](/uploads/acb972b79ee5bbc271189695ca11f4d6/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Jan 3, 2020, 23:31

@kdjstudios @jredfield 

I mixed up the CASE headings in the above comment, I apologize for that.

**CASE 1:** Use Parent Account Activities is set as `True` and  Separate child accounts transactions is set as `False`

In this case, it will sum all the parent & child account activities usage value and it will treat it as a single transaction. For Eg.

![image](/uploads/36ed7de0d28b9546931e5e9335a94663/image.png)

**CASE 2:** Use Parent Account Activities is set as `True` and  Separate child accounts transactions is set as `True`

In this case, it will treat all the activities of the parent and child accounts as separate transactions.
But to calculate the quantity on the child accounts, it will take base rate and allowance of the parent account activities.

For Eg. 

* Account `071-1234` actual quantity is **193** but its allowance is **700**, so its billable is **0**
* Account `071-1237` actual quantity is **37** but its allowance is **700**, so its billable is **0**

![image](/uploads/af917637293952564d10830167cfb104/image.png)

**In short, they had separate `true` at the time of invoice creation.**

**The example mentioned in #1697 by Leah Mitchell, is the scenario of example 3 mentioned in the [Test cases](https://docs.google.com/spreadsheets/d/1mQVZ_ra4uNvHbcdMsX9Phu9uCPyF3aq8Xr6dNuKevC4/edit?usp=sharing) and here are the final draft pdfs for [Case 1(Bill To `True`, separate `False`) & Case 4(Both as `True`)](http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1362#note_41385) of #1362. And it is working according to that.**

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 6, 2020, 08:28

This was reviewed today during a call. We discussed that the expected behavior of when both are marked true is to have all usage for all accounts be moved to the parent and charged based upon the parent account activity. While we still will separate out the (non usage) charges into their child account individually. The Aavaz team will put together a recap of their understanding and and estimate to develop this update/change.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 6, 2020, 12:33

I wrote:

> We were able to review further in depth on this with the support team and we believe we found where the disconnect is occurring in SA Billing. The team is compiling a bit more research on this before we present anything but I did want to give you an update.

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Jan 7, 2020, 03:59

@kdjstudios @jredfield 

As per our discussion, I have created an excel sheet for the layout of the invoice when both **use parent activities** and **Separate child accounts transactions** are `True`. Only recurring fee line items are printed as separate line items for parent-child accounts and all the usage are added to the parent account and only one line item per activity is printed under the parent account.

[parent_child.xlsx](/uploads/9df5aaf8fc769e0bdde9a003b3d8af37/parent_child.xlsx)

**Please review this and give your feedback on it.**

NOTE: This is looking exactly like the invoice when use parent activities is `True` and Separate child accounts transactions is `False`.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 7, 2020, 12:01

@ranand 

Yes that looks correct to me for base rates (reoccurring fees). NOTE: I would like to confirm how staged fees, holiday fees, late fees, additional fees, etc. will also look too when added to both child and/or parent.

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Jan 8, 2020, 02:50

@kdjstudios @jredfield 

Now we've added the staged fees, holiday fees, late fee, SD, & taxes into the invoice format. Please review the attached updated invoice format. I hope we've covered all kinds of line items we can add to the invoice. If you can think of any other type of line item which needs to be added to the invoice then please do let us know. 

[parent_child.xlsx](/uploads/59a8e71b47710b156c9df2db94191461/parent_child.xlsx)

Please give your feedback on it.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 8, 2020, 11:33

@ranand @jredfield 

Looks like the child account late charges not being charged? They should be handled the same way we handle staged fees on the child accounts in this scenario (Both True). 

Looks like the holiday charges are charging the wrong amounts? They should be handled the same way we handle staged fees on the child accounts in this scenario (Both True). Also we need to keep in mind if the Holiday charge is "enabled" on each account too - #1696 . 

@jredfield We may want to review this with Gary to make sure my understanding of this is the same as his wants.

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Jan 9, 2020, 03:14

@kdjstudios 

> Looks like the child account late charges not being charged? They should be handled the same way we handle staged fees on the child accounts in this scenario (Both True).

As per the current functionality of the SAB, Late charges did not get affected by Bill To and separate options, as late charges are based on the total AR balance of the account on which we are creating the invoice. It will always look for the late charges on the parent account if there is any AR balance.


> Looks like the holiday charges are charging the wrong amounts? They should be handled the same way we handle staged fees on the child accounts in this scenario (Both True). Also we need to keep in mind if the Holiday charge is "enabled" on each account too - #1696 .

Please elaborate on this, because as per my understanding if **Bill To** is `True` and **Separate** as `True` and parent & all child accounts have **holiday charge** set as `True` then Parent account & all child accounts will have holiday charges on them as separate line items, just the applied holiday charge amount will be the amount which is set on the parent account.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Jan 9, 2020, 21:52

@kdjstudios @jredfield 

updating label as this ticket is pending for Gary's review.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 14, 2020, 10:02

@jredfield @ranand 

This was reviewed with Gary today. We will need to fix the billing to behave like you have outlined in http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1679#note_54157. I will also want to work on updating the documentation for these options to better describe and visualize the different scenarios. 

Ticket #1696 is addressing how holiday charges should be billed, so we can leave that conversation on that ticket. 

@jredfield please feel free to add any additional information I may have missed. EDIT: I also updated the priority on this since we will want this updated and released before the sites start to process their next billing.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Jan 14, 2020, 23:36

@kdjstudios @jredfield 

Thank you for the update.

We have completed the development & setup branch on the dev server for your testing. Please test it.

Our testing results:

* Usage File: [VCC_test_export](/uploads/871312f34e345ffeac1800f16065f958/VCC_test_export_bill_to.csv)

* ![image](/uploads/51dfd87729254a042475d88cd5af472b/image.png)

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 15, 2020, 11:55

This looks good marking for announcement.

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Jan 18, 2020, 24:00

@kdjstudios @jredfield 

We have successfully released this on both the production servers.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 18, 2020, 11:57

I wrote:

> The support team was able to release the update to SA Billing to adjust how parent child accounts generate their invoices when both the "Use Parent Activities" and "Separate Child Account Transactions" is selected. Please pay additional attention to your parent/child accounts on your upcoming billing cycles to confirm everything is working as expected. Thank you and we look forward to your response.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jan 23, 2020, 15:35

closed
